yt_dlp/downloader/mhtml.py

```diff
@@ -0,0 +1,175 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .fragment import FragmentFD
+import uuid
+import io
+from ..version import __version__ as YT_DLP_VERSION
+from ..utils import (
+    urljoin,
+    srt_subtitles_timecode,
+    formatSeconds,
+)
+
+
+class MhtmlFD(FragmentFD):
+    FD_NAME = 'mhtml'
+    _EXTENSION = 'mhtml'
+
+    _STYLESHEET = """\
+html, body {
+    margin: 0;
+    padding: 0;
+    height: 100vh;
+}
+
+html {
+    overflow-y: scroll;
+    scroll-snap-type: y mandatory;
+}
+
+body {
+    scroll-snap-type: y mandatory;
+    display: flex;
+    flex-flow: column;
+}
+
+body > figure {
+    max-width: 100vw;
+    max-height: 100vh;
+    scroll-snap-align: center;
+}
+
+body > figure > figcaption {
+    text-align: center;
+    height: 2.5em;
+}
+
+body > figure > img {
+    display: block;
+    margin: auto;
+    max-width: 100%;
+    max-height: calc(100vh - 5em);
+}
+"""
+
+    def _gen_cid(self, i, fragment, frag_boundary):
+        return '%u.%s@yt-dlp.github.io.invalid' % (i, frag_boundary.decode('us-ascii'))
+
+    def _gen_stub(self, fragments, frag_boundary):
+        output = io.StringIO()
+
+        output.write(
+            '<!DOCTYPE html>'
+            '<html>'
+            '<head>'
+            ''  '<meta name="generator" content="yt-dlp %s">'
+            ''  '<title>yt-dlp downloaded slideshow</title>'
```

Comment:
I thought about it too, but decided to submit an MVP version first, since I didn’t want to become too distracted by improvements to this ‘player’ page. It does seem like a good idea, though.